sklearn/metrics/pairwise.py

```diff
@@ -256,6 +284,123 @@ def euclidean_distances(X, Y=None, Y_norm_squared=None, squared=False,
     return distances if squared else np.sqrt(distances, out=distances)
 
 
+def masked_euclidean_distances(X, Y=None, squared=False,
+                               missing_values="NaN", copy=True):
+    """Calculates euclidean distances in the presence of missing values
+
+    Considering the rows of X (and Y=X) as samples, compute the distance matrix
+    between each pair of samples. Similarly, if Y is not X, then compute the
+    distance matrix between each sample pair (i.e., each row pair) in X and Y.
+
+    When calculating the distance between a pair of samples, this formulation
+    essentially zero-weights feature coordinates with a missing value in either
+    sample and scales up the weight of the remaining coordinates:
+
+        dist(x,y) = sqrt(weight * sq. distance from non-missing coordinates)
+        where,
+        weight = Total # of coordinates / # of non-missing coordinates
+
+    Note that if all the coordinates are missing or if there are no common
```

Comment:
Isn't the first just a special case of the second?

Reply:
But I don't think the description distracts from the point we are trying to make, and in fact explicitly lays out what would happen in the "special case" scenario. I feel it is worthwhile to keep, what do you think?